src/test/java/org/prebid/server/bidder/adrino/AdrinoBidderTest.java

```diff
@@ -0,0 +1,299 @@
+                        .build())
+                .device(Device.builder()
+                        .w(1920)
+                        .h(800)
+                        .build())
+                .build();
+
+        final BidderCall<BidRequest> httpCall = givenHttpCall(response);
+
+        // when
+        final Result<List<BidderBid>> result = adrinoBidder.makeBids(httpCall, bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1)
+                .containsExactly(BidderBid.of(Bid.builder()
+                        .id("impId")
+                        .impid("test-imp-id")
+                        .price(BigDecimal.TEN)
+                        .adm("{json response string}")
+                        .ext(mapper.valueToTree(
+                                ExtPrebid.of(ExtBidPrebid.builder().type(BidType.xNative).build(), null)))
+                        .crid("test-creative-id").mtype(4).build(), BidType.xNative, "PLN"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnEmptyResponseWhenWhereIsNoBid() throws JsonProcessingException {
+        // given
+        final String response = mapper.writeValueAsString(null);
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder().id("impId").build()))
+                .build();
+
+        final BidderCall<BidRequest> httpCall = givenHttpCall(response);
+
+        // when
+        final Result<List<BidderBid>> result = adrinoBidder.makeBids(httpCall, bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldReturnBidsFromDifferentSeatBidsInResponse() throws JsonProcessingException {
+        // given
+        final String response = mapper.writeValueAsString(BidResponse.builder()
+                .seatbid(asList(
+                        SeatBid.builder()
+                                .bid(singletonList(Bid.builder().id("bidId1").impid("impId1").build()))
+                                .build(),
+                        SeatBid.builder()
+                                .bid(singletonList(Bid.builder().id("bidId2").impid("impId2").build()))
+                                .build()))
+                .build());
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(asList(Imp.builder().id("impId1").build(), Imp.builder().id("impId2").build()))
+                .build();
+
+        final BidderCall<BidRequest> httpCall = givenHttpCall(response);
+
+        // when
+        final Result<List<BidderBid>> result = adrinoBidder.makeBids(httpCall, bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(2)
+                .extracting(BidderBid::getBid)
+                .extracting(Bid::getId).containsExactly("bidId1", "bidId2");
```

Comment:
In that case, we can drop `hasSize` assertion. `containsExactly` already implies that.